File: app/src/main/java/github/umer0586/sensorserver/util/SensorUtil.java
package github.umer0586.sensorserver.util;

import android.content.Context;
import android.hardware.Sensor;
import android.hardware.SensorManager;

import java.util.List;

public class SensorUtil {

    private static SensorUtil utilInstance = null;
    private SensorManager sensorManager;

    private SensorUtil(Context context)
    {
        sensorManager = (SensorManager) context.getSystemService(context.SENSOR_SERVICE);
    }

    public static synchronized SensorUtil getInstance(Context context)
    {

        if(utilInstance == null)
            utilInstance  = new SensorUtil(context);

        return utilInstance;

    }

    public List<Sensor> getAvailableSensors()
    {
        return sensorManager.getSensorList(Sensor.TYPE_ALL);
    }

    public Sensor getSensorFromStringType(String sensorStringType)
    {

        for(Sensor availableSensor : getAvailableSensors())
            if(availableSensor.getStringType().equalsIgnoreCase(sensorStringType))
                return availableSensor;


         return null;
    }


}

File: app/src/main/java/github/umer0586/sensorserver/util/SensorUtil.kt
package github.umer0586.sensorserver.util

import android.content.Context
import android.hardware.Sensor
import android.hardware.SensorManager

class SensorUtil private constructor(context: Context?)
{


    private val sensorManager: SensorManager

    init
    {
        sensorManager = context!!.getSystemService(Context.SENSOR_SERVICE) as SensorManager
    }

    val availableSensors: List<Sensor>
        get() = sensorManager.getSensorList(Sensor.TYPE_ALL)

    fun getSensorFromStringType(sensorStringType: String?): Sensor?
    {
        for (availableSensor in availableSensors) if (availableSensor.stringType.equals(
                sensorStringType,
                ignoreCase = true
            )
        ) return availableSensor
        return null
    }

    companion object
    {


        private var utilInstance: SensorUtil? = null

        @kotlin.jvm.Synchronized
        fun getInstance(context: Context?): SensorUtil?
        {
            if (utilInstance == null) utilInstance = SensorUtil(context)
            return utilInstance
        }
    }
}